prometheus/promhttp/http.go

```diff
@@ -163,22 +168,38 @@ func HandlerFor(reg prometheus.Gatherer, opts HandlerOpts) http.Handler {
 		enc := expfmt.NewEncoder(w, contentType)
 
 		var lastErr error
+
+		// handleError returns true if we have to abort after handling the error.
+		handleError := func(err error) bool {
+			if err == nil {
+				return false
+			}
+			lastErr = err
+			if opts.ErrorLog != nil {
+				opts.ErrorLog.Println("error encoding and sending metric family:", err)
+			}
+			errCnt.WithLabelValues("encoding").Inc()
+			switch opts.ErrorHandling {
+			case PanicOnError:
+				panic(err)
+			case ContinueOnError:
+				// Handled later.
```

Comment:
Returning here would make more sense now with the refactoring.

Reply:
I thought so, too, at first. But it would be redundant with the return in L191. I had it first there, and also had a `default` clause to explicitly handle all possibilities (which then doesn't require the final `return` in the function anymore), but then thought that's also redundant.

On the other hand, to really do the shortest possible code, the `ContinueOnError` could be completely omitted. I did that now, with a short comment.